src/Compilers/CSharp/Portable/Symbols/Source/SourceTypeParameterSymbol.cs

```diff
@@ -256,18 +256,15 @@ private void CheckConstraintTypeConstraints(DiagnosticBag diagnostics)
                 return;
             }
 
-            var corLibrary = this.ContainingAssembly.CorLibrary;
-            var conversions = new TypeConversions(corLibrary);
-            var location = _locations[0];
-
+            var args = new CheckConstraintsArgs(DeclaringCompilation, new TypeConversions(ContainingAssembly.CorLibrary), _locations[0], diagnostics);
             foreach (var constraintType in constraintTypes)
             {
                 HashSet<DiagnosticInfo> useSiteDiagnostics = null;
                 constraintType.Type.AddUseSiteDiagnostics(ref useSiteDiagnostics);
 
-                if (!diagnostics.Add(location, useSiteDiagnostics))
+                if (!diagnostics.Add(args.Location, useSiteDiagnostics))
                 {
-                    constraintType.Type.CheckAllConstraints(DeclaringCompilation, conversions, location, diagnostics);
+                    constraintType.Type.CheckAllConstraints(args);
```

Comment:
Are these all overloads that already exist (accepting a CheckConstraintsArgs instead of the individual values)? And so the value is in removing the intermediate call from the stack loop?

Reply:
The signature of the method was changed to take `in CheckConstraintsArgs` rather than separate parameters. The goal was to allow the same `struct` to be passed by reference to several nested calls.

---
In reply to: [405100793](https://github.com/dotnet/roslyn/pull/43094#discussion_r405100793) [](ancestors = 405100793)

Reply:
Oops, I did not observe that these were all extension methods.